### PR TITLE
Extra Field Order Struct

### PR DESCRIPTION
**Pull request by ejwessel**
## Description
https://github.com/airswap/airswap-protocols/issues/367

## Changes

- rename `param` to `amount`
- add `id` to `Party` struct

## Tests

### CODE REVIEW

**Comment by ejwessel** [comment]
not actually done. I need to finish handling `swap.transferToken`

File: source/delegate/contracts/Delegate.sol [hunk 1]
```diff
@@ -298,7 +298,7 @@ contract Delegate is IDelegate, Ownable {
       // Ensure the senderParam does not exceed maximum for the rule.
       if(senderParam <= rule.maxSenderAmount) {
```
Comment (poiresel) [inline source/delegate/contracts/Delegate.sol]: senderAmount instead of senderParam

File: source/delegate/contracts/Delegate.sol [hunk 2]
```diff
@@ -298,7 +298,7 @@ contract Delegate is IDelegate, Ownable {
...
 
-        signerParam = _calculateSignerParam(senderParam, rule.priceCoef, rule.priceExp);
+        signerParam = _calculateSignerAmount(senderParam, rule.priceCoef, rule.priceExp);
 
         // Return the quote.
         return signerParam;
```
Comment (poiresel) [inline source/delegate/contracts/Delegate.sol]: signerAmount

File: source/delegate/contracts/Delegate.sol [hunk 3]
```diff
@@ -328,7 +328,7 @@ contract Delegate is IDelegate, Ownable {
     if(rule.maxSenderAmount > 0) {
 
       // Calculate the senderParam.
-      senderParam = _calculateSenderParam(signerParam, rule.priceCoef, rule.priceExp);
+      senderParam = _calculateSenderAmount(signerParam, rule.priceCoef, rule.priceExp);
```
Comment (poiresel) [inline source/delegate/contracts/Delegate.sol]: signerAmount

File: source/delegate/contracts/Delegate.sol [hunk 4]
```diff
@@ -328,7 +328,7 @@ contract Delegate is IDelegate, Ownable {
...
 
       // Ensure the senderParam does not exceed the maximum trade amount.
```
Comment (poiresel) [inline source/delegate/contracts/Delegate.sol]: senderAmount

**Comment by ejwessel** [comment]
seems MIGRATION.md needs to be updated entirely. Let's do this in another PR.

File: source/delegate/contracts/Delegate.sol [hunk 5]
```diff
@@ -429,42 +429,42 @@ contract Delegate is IDelegate, Ownable {
...
+    signerAmount = multiplier.div(10 ** priceExp);
 
     // If the div rounded down, round up
     if (multiplier.mod(10 ** priceExp) > 0) {
-      signerParam++;
+      signerAmount++;
     }
   }
 
   /**
```
Comment (poiresel) [inline source/delegate/contracts/Delegate.sol]: param in the natspec

File: source/swap/contracts/Swap.sol [hunk 6]
```diff
@@ -329,14 +343,16 @@ contract Swap is ISwap {
     * @dev ERC20: Takes into account non-standard ERC-20 tokens.
     * @param from address Wallet address to transfer from
     * @param to address Wallet address to transfer to
-    * @param param uint256 Amount for ERC-20 or token ID for ERC-721
+    * @param amount uint256 Amount for ERC-20 or token ID for ERC-721
```
Comment (dmosites) [inline source/swap/contracts/Swap.sol]: "Amount for ERC-20"

File: source/types/contracts/Types.sol [hunk 7]
```diff
@@ -37,7 +37,8 @@ library Types {
     bytes4 kind;                  // Interface ID of the token
     address wallet;               // Wallet address of the party
     address token;                // Contract address of the token
-    uint256 param;                // Value (ERC-20) or ID (ERC-721)
+    uint256 amount;               // Amount of (ERC-20) or (ERC-721)
```
Comment (dmosites) [inline source/types/contracts/Types.sol]: "Amount for ERC-20 or ERC-1155"

File: source/types/contracts/Types.sol [hunk 8]
```diff
@@ -37,7 +37,8 @@ library Types {
...
+    uint256 id;                   // ID of ERC-721 token; 0 for ERC-20
```
Comment (dmosites) [inline source/types/contracts/Types.sol]: "ID for ERC-721 or ERC-1155"

**Pull request merged**